app/models/flag.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class Flag < ActiveRecord::Base
+  VALID_APPLIES_TO = %w[Post Topic Chat::Message]
```

Comment:
Oh sorry I am going to be a huge pain again...since this is a plugin model, we need some plugin registry way of registering this instead of hardcoding it in core. A good example of how to do this is bookmarks:

https://github.com/discourse/discourse/blob/5cd58e7aab15414b588cd53348a6c14e55c08e93/app/models/bookmark.rb#L4-L11

So you would have `DEFAULT_VALID_APPLIES_TO = %w[Post Topic]` then have

```ruby
  def self.valid_applies_to_types
    Set.new(DEFAULT_VALID_APPLIES_TO | DiscoursePluginRegistry.flag_applies_to_types)
  end
```

And register the Chat::Message model in chat